kitsune/search/v2/documents.py

```diff
@@ -170,14 +186,38 @@ def prepare_is_solution(self, instance):
     def prepare_locale(self, instance):
         return instance.question.locale
 
+    def prepare_num_helpful_votes(self, instance):
+        if hasattr(instance, "es_num_helpful_votes"):
+            return instance.es_num_helpful_votes
+        return instance.num_helpful_votes
+
+    def prepare_num_unhelpful_votes(self, instance):
+        if hasattr(instance, "es_num_unhelpful_votes"):
+            return instance.es_num_unhelpful_votes
+        return instance.num_unhelpful_votes
+
     def get_field_value(self, field, instance, *args):
         if field.startswith("question_"):
             instance = instance.question
         return super().get_field_value(field, instance, *args)
 
     @classmethod
     def get_model(cls):
-        return question_models.Answer
+        return Answer
+
+    @classmethod
+    def get_queryset(cls):
+        return (
+            Answer.objects
+            # prefetch each answer's question,
+            # applying the same optimizations as in the QuestionDocument
+            .prefetch_related(Prefetch("question", queryset=super().get_queryset()))
+            # count votes in db to improve performance
+            .annotate(
+                es_num_helpful_votes=Count("votes", filter=Q(votes__helpful=True)),
```

Comment:
Why do we need the `Q` object here? Same question for the next line

Reply:
This is needed to filter by only votes that are helpful/not helpful when counting. The generated SQL might help:
```
COUNT(CASE WHEN `questions_answervote`.`helpful` = True THEN `questions_answervote`.`id` ELSE NULL END) AS `es_num_helpful_votes`, COUNT(CASE WHEN `questions_answervote`.`helpful` = False THEN `questions_answervote`.`id` ELSE NULL END) AS `es_num_unhelpful_votes`
```

There are a lot of gotchas with annotations, so I want to add some basic unit tests here to ensure we're not double counting votes or anything.